### PR TITLE
Rename LastProcessedEvent to LastWorkflowTaskStartId

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Rename LastProcessedEvent to LastWorkflowTaskCompleteId since LastProcessedEvent is misleading

**Why?**
See above

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/historyEngine2_test.go [hunk 1]
```diff
@@ -212,8 +212,8 @@ func (s *engine2Suite) TestRecordWorkflowTaskStartedSuccessStickyExpired() {
 	expectedResponse := historyservice.RecordWorkflowTaskStartedResponse{}
 	expectedResponse.WorkflowType = msBuilder.GetWorkflowType()
 	executionInfo = msBuilder.GetExecutionInfo()
-	if executionInfo.LastProcessedEvent != common.EmptyEventID {
-		expectedResponse.PreviousStartedEventId = executionInfo.LastProcessedEvent
+	if executionInfo.LastWorkflowTaskCompleteId != common.EmptyEventID {
+		expectedResponse.PreviousStartedEventId = executionInfo.LastWorkflowTaskCompleteId
```
Comment (yiminc) [inline service/history/historyEngine2_test.go:216]: is LastProcessedEvent always be WFT completed event? Could it be ActivityTaskCompleted or anything else?
Here what is the PreviousStartedEventId? It does not sound right that you set a "StartedEventId" as WFTCompleted event id.

Comment (wxing1292) [reply service/history/historyEngine2_test.go:216]: https://github.com/temporalio/temporal/blob/2666ad925b2d627a622b3a0fbb928d5b2bea4572/service/history/workflow/workflow_task_state_machine.go#L744

not sure about the `PreviousStartedEventId`

Comment (yiminc) [reply service/history/historyEngine2_test.go:216]: From the line you quote, the LastProcessedEvent is the WorkflowTaskStartedEvent (not the completed event).

Comment (wxing1292) [reply service/history/historyEngine2_test.go:216]: > From the line you quote, the LastProcessedEvent is the WorkflowTaskStartedEvent (not the completed event).

good catch

**Review by dtornow — APPROVED** [review]

**Review by yiminc — APPROVED** [review]

**Pull request merged**